### PR TITLE
Shipping Labels: networking layer for supporting the `data/countries` endpoint

### PR DESCRIPTION
**Pull request by pmusolino**
Part of #4464 

## Description
In this PR, I implemented the remote endpoint [data/countries](https://woocommerce.github.io/woocommerce-rest-api-docs/#retrieve-continent-data) that will be used to fetch all the countries and the states, which will be implemented in the Shipping Label Address Validation screen. The Yosemite/Storage layers will be implemented in future PRs.
Note: more than 300 LOC are because of the JSON file `countries.json`.

## Testing
- Just CI, for the moment the endpoint is not used in the rest of the codebase.

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Fakes/Fakes/Fakes.generated.swift [hunk 1]
```diff
@@ -1294,6 +1305,16 @@ extension StatGranularity {
         .day
     }
 }
+extension StateOfACountry {
+    /// Returns a "ready to use" type filled with fake values.
```
Comment (allendav) [inline Fakes/Fakes/Fakes.generated.swift:1309]: Since "state" might be an ambiguous or unfamilar term to some maintainers, would you mind adding a comment along the lines of "with fake values for a state/province of a county"

File: Fakes/Fakes/Fakes.generated.swift [hunk 2]
```diff
@@ -1360,3 +1381,17 @@ extension UploadableMedia {
         )
     }
 }
+extension WCPayAccountStatusEnum {
```
Comment (allendav) [inline Fakes/Fakes/Fakes.generated.swift:1384]: Why these changes?

File: Networking/Networking/Model/Country.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import Foundation
+import Codegen
+
+/// Represents a Country Entity.
+///
+public struct Country: Decodable, Equatable, GeneratedFakeable {
+    public let code: String
```
Comment (allendav) [inline Networking/Networking/Model/Country.swift:9]: Optional. Add an example as a comment, e.g.

// ISO-3166 two letter country code, e.g. CA for Canada

File: Networking/Networking/Model/StateOfACountry.swift [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import Foundation
+import Codegen
+
+/// Represents a State Entity within a Country.
+///
+public struct StateOfACountry: Decodable, Equatable, GeneratedFakeable {
+    public let code: String
```
Comment (allendav) [inline Networking/Networking/Model/StateOfACountry.swift:9]: Optional. Add an example as a comment, e.g.:

// E.g., ON for Ontario. Note: Not always two letters.

File: Networking/NetworkingTests/Responses/countries.json [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+                ],
+                "collection": [
+                    {
+                        "href": "https://mywootesting.mystagingwebsite.com/wp-json/wc/v3/data/countries"
+                    }
+                ]
+            }
+        }
+    ]
+}
```
Comment (allendav) [inline Networking/NetworkingTests/Responses/countries.json:345]: Some countries don't have states or provinces. For completeness, the unit test should include at least one such country, e.g. I think Austria (AT) might work

File: Fakes/Fakes/Fakes.generated.swift (hunk 1 above)
Comment (allendav) [reply Fakes/Fakes/Fakes.generated.swift:1309]: Or, is this not possible since this is generated code?

**Review by allendav — APPROVED** [review]
A few questions/suggestions to consider, especially to add a country without states to the unit test. `test_countries_are_properly_parsed` passes fine for me

File: Fakes/Fakes/Fakes.generated.swift (hunk 1 above)
Comment (pmusolino) [reply Fakes/Fakes/Fakes.generated.swift:1309]: This is generated code, so if I edit the file, the next time that someone else generates new fakes, my changes will be deleted. 👍

File: Fakes/Fakes/Fakes.generated.swift (hunk 2 above)
Comment (pmusolino) [reply Fakes/Fakes/Fakes.generated.swift:1384]: I think that someone added the `GeneratedFakeable` conformance to those models, but never generated the Fakes extensions. :nod:

File: Networking/NetworkingTests/Responses/countries.json (hunk 5 above)
Comment (pmusolino) [reply Networking/NetworkingTests/Responses/countries.json:345]: Good idea 👍 done

**Comment by pmusolino** [comment]
Thank you @allendav for the review! I responded to all your feedback and update the code where needed, including a country without states.
